Bloggist/src/triggers/statusChangeTrigger.trigger

```diff
@@ -0,0 +1,9 @@
+trigger statusChangeTrigger on Post__c (before update) {
+    for(Post__c p : Trigger.New) {
+        if(p.Status__c == 'Draft') {
+            p.Status__c = 'Under Review';
```

Comment:
Why is this trigger automatically switching and flipping statuses?

Reply:
The idea here is that the post at this point is being updated only when is changing its status. The only way of making this is through clicking buttons for submitting (which changes the status from draft to under review) approving (which changes the status from under review to ready). The same way i was thinking of adding later on another change from ready to published when the button for publish is clicked. Also another rules can be added to manage the update depending on the source of the action. If not a good idea it can be changed to an approach were the processing of the status is made on the js controllers instead of a trigger on the server side.

Reply:
It's not a bad idea to do certain operations in triggers, the problem is actually what this trigger does.
In salesforce, there's absolutely no way for you to limit the way data is generated. You can never assert users will create posts through your UI or even using a UI at all. 
Same thing happens with updates so this trigger will execute under any update on any post record. 
Also, what this trigger does in reality is probably not what you meant: it switches any post that gets update and is in "Draft" status to "Under Review", regardless of the type of change performed.